src/disposeOnUnmount.js

```diff
@@ -0,0 +1,52 @@
+import * as React from "react"
+import { patch } from "./utils/utils"
+
+const storeKey = "__$mobxDisposeOnUnmount"
+
+function customComponentWillUnmount() {
+    this[storeKey].forEach(propKeyOrFunction => {
```

Comment:
NIT: `this[storeKey].splice(0).forEach` prevents concurrent modifications, and doesn't need the reset of line 19. (No need to change it if there are no other changes for this MR, just a nice pattern in general, imho)

Reply:
OK, actually forEach does, but (for of) doesn't :)
still, since this once uses forEach it should be ok